sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala

```diff
@@ -56,7 +56,14 @@ case class DataSourceV2Relation(
     scan match {
       case r: SupportsReportStatistics =>
         val statistics = r.estimateStatistics()
-        Statistics(sizeInBytes = statistics.sizeInBytes().orElse(conf.defaultSizeInBytes))
+        val numRows: Option[BigInt] = if (statistics.numRows().isPresent) {
+          Some(statistics.numRows().getAsLong)
+        } else {
+          None
+        }
+        Statistics(
```

Comment:
shall we create a util method to turn v2 statistics to spark statistics?

Reply:
OK, will add it later.